bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/extensions/ExtensionResource.java

```diff
@@ -149,6 +153,25 @@ public void run() {
     }
 
     @POST
+    @Path("/url/{url}/install")
+    @ApiOperation(value = "Installs the extension from the given URL.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 400, message = "The given URL is malformed or not valid.") })
+    public Response installExtensionByURL(
+            final @PathParam("url") @ApiParam(value = "extension install URL", required = true) String url) {
+        try {
+            URI extensionURI = new URI(url);
+            String extensionId = getExtensionId(extensionURI);
+            installExtension(extensionId);
+        } catch (Exception e) {
```

Comment:
yes, `new URI(url)` will throw an URISyntaxException in case the given string violates RFC 2396. This was meant to be reported by `given URL is malformed`.